src/Mono.Android/Android.Runtime/JNIEnv.cs

```diff
@@ -477,10 +477,13 @@ public static string GetJniName (Type type)
 			if (type == null)
 				throw new ArgumentNullException ("type");
 
-			string? java = TypemapManagedToJava (type);
-			return java == null
+			JniTypeSignature sig = default;
+			if (!type.ContainsGenericParameters)
+				sig = JNIEnvInit.androidRuntime?.TypeManager.GetTypeSignature (type) ?? default;
+
+			return sig.SimpleReference == null
 				? JavaNativeTypeManager.ToJniName (type)
-				: java;
+				: sig.Name;
```

Comment:
I think this needs separate handling for the case when `sig is null`.

Reply:
`JniTypeSignature` is a struct, and I found other examples that check `SimpleReference` to see if it is an "empty" struct:

* https://github.com/dotnet/java-interop/blob/c1cd06dde871b468f6c729fdce496d2971b95003/src/Java.Interop/Java.Interop/JavaObjectArray.cs#L27-L29

Reply:
I see. Thanks for the clarification.